python/pyarrow/array.pxi

```diff
@@ -218,6 +218,29 @@ def asarray(values, type=None):
         return array(values, type=type)
 
 
+def infer_type(value, from_pandas=False):
```

Comment:
minor nit, but maybe `value` -> `values`. To be consistent with `asarray` above, and to be clear we expect an array-like of values, not a single value (scalar).

Reply:
done